infra/modules/providers/azure/app-service/main.tf

```diff
@@ -0,0 +1,13 @@
+resource "azurerm_resource_group" "appsvc" {
```

Comment:
why are we defining a resource group here as opposed to referencing the one that contains the app service plan resource?

Reply:
@erikschlegel: When we reference a resource group at resource level, and we have a case of multiple resources referencing the same resource, terraform deletes the resource and creates a new resource. But if its done at a module level, it works as expected. We did a couple different experimental runs at azure-simple level as bedrock is doing it and it works perfectly as expected. Thus we have to define the resource group at this level as opposed to referencing it.